dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java

```diff
@@ -210,6 +215,15 @@ private String buildIconQuery(
       parameterSource.addValue("keys", params.getKeys());
     }
 
+    if (params.hasSearch()) {
+      String searchValue = params.getSearch();
+
+      sql += hlp.whereAnd() + " c.iconkey ilike :search";
+      sql += " OR jsonb_path_exists(keywords, '$[*] ? (@ like_regex " + doubleQuote(searchValue) + ")')";
+
+      parameterSource.addValue("search", wrap(addIlikeReplacingCharacters(searchValue), '%'));
```

Comment:
not sure if I should use `addIlikeReplacingCharacters`, but it seemed the safest choice. However this is in the `dataitem`-package, I'm not sure how strict you are on including code from other modules like that?

Reply:
You might have to add the module dependency if it wasn't already there

Reply:
Yeah, vscode took care of that